### PR TITLE
feat(`NextSteps`): replace `Intro` CTAs with new CTA, change tile colors and enable badges

### PR DESCRIPTION
**Pull request by nandereck**
🎟️ [Asana Task](https://app.asana.com/0/1202738955551191/1203512370732554/f)
🔍 [Preview Link](https://react-components-git-nanext-steps-updates-hashicorp.vercel.app/components/nextsteps?isFullscreen=true)

---

## Description
This PR updates the `NextSteps` component according to updates in [the spec](https://www.figma.com/file/7cYgDM618stjYUHDqAfRec/Components?node-id=4757%3A31522&t=aZPFJr9zBpNwoozj-1). The changes are as follows:
- enables optional `Badge` in tiles
- enables optional tertiary CTA
- updates tiles to use `secondary` colors
- updates layout according to spec

## Validation Steps
- Navigate to `NextSteps` in the [Vercel preview deploy](https://react-components-git-nanext-steps-updates-hashicorp.vercel.app/components/nextsteps?isFullscreen=true)
- Validate that the `Badge` implementation matches [the spec](https://www.figma.com/file/7cYgDM618stjYUHDqAfRec/Components?node-id=4757%3A31522&t=aZPFJr9zBpNwoozj-1).
- Scroll down to the `Optional Badge` example in the examples section
- Validate that the appropriate variant of the `Badge` renders
- Validate that the secondary color for Consul is used for the tile background
- Scroll down to the `Optional Tertiary CTA` example in the examples section
- Validate that it matches the spec

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Comment by nandereck** [comment]
Still waiting on design feedback on how we will handle the CTA in `Intro` and the new tertiary CTA. Based on the current draft in design, it looks like we will remove the `Intro` CTA in favor of the new one. Either way, will wait until that gets finalized by design before merging this in.

File: packages/next-steps/index.tsx
Comment (alexcarpenter) [inline packages/next-steps/index.tsx:115]: Were we able to keep the badge inline with the text when wrapping? Or what is the expected behavior here?

CleanShot 2023-01-18 at 15 05 20@2x

Comment (nandereck) [reply packages/next-steps/index.tsx:115]: Oops - I think you flagged this before. Think this does the trick:

Screen Shot 2023-01-18 at 12 32 33 PM

**Comment by nandereck** [comment]
@alexcarpenter - confirmed with design that we're good to go. Mind giving this a review?

File: packages/next-steps/docs.mdx
Comment (alexcarpenter) [inline packages/next-steps/docs.mdx:98]: Confused, should this be tertiary CTA?

File: packages/next-steps/types.ts
Comment (alexcarpenter) [inline packages/next-steps/types.ts:50]: So we are keeping the cta option but limiting to one?

**Review by alexcarpenter — COMMENTED** [review]
@nandereck Looks like we need to remove tertiaryCtas from types?

File: packages/next-steps/docs.mdx
Comment (alexcarpenter) [reply packages/next-steps/docs.mdx:98]: nvm, I see we just need to remove the extra type.

File: packages/next-steps/types.ts
Comment (nandereck) [reply packages/next-steps/types.ts]: Oops! `tertiaryCta` was a placeholder when I wasn't sure if we were still enabling CTAs in the `Intro` component.

I've removed references to `tertiaryCta` and updates both the component's types and props files.

**Review by alexcarpenter — APPROVED** [review]
:shipit:

**Pull request merged**
